### PR TITLE
Added QueueClient.CreateIfNotExists(), .Exists(), add .DeleteIfExists()

### LINKED ISSUE #7879: [FEATURE REQ] CreateIfNotExists/DeleteIfExists/Exists for Queue
**Is your feature request related to a problem? Please describe.**
Related to issue #7211 
In the old API there are 'convenience' methods like CreateIfNotExists/DeleteIfExists for File and FileShare

**Describe the solution you'd like**
Helper methods (like CreateIfNotExists/DeleteIfExists) for Files

**Describe alternatives you've considered**
Can easily be worked around. Refer to #7211.

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Description Added
- [x] Expected solution specified

### PR DESCRIPTION
**Pull request by seanmcc-msft**
Resolves https://github.com/Azure/azure-sdk-for-net/issues/7879

### CODE REVIEW

**Comment by seanmcc-msft** [comment]
@tg-msft, I'm thinking maybe we could potentially start grouping our optional parameters in this PR.  What do you think?

File: sdk/storage/Azure.Storage.Queues/src/QueueClient.cs [hunk 1]
```diff
@@ -411,6 +417,352 @@ private void SetNameFieldsIfNull()
         }
         #endregion Create
 
+        #region Create If Not Exists
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Queues/src/QueueClient.cs]: NIT - remove spaces. Also I believe for blobs, I put the `CreateIfNotExists` methods in the `Create` region

File: sdk/storage/Azure.Storage.Queues/src/QueueClient.cs [hunk 2]
```diff
@@ -411,6 +417,352 @@ private void SetNameFieldsIfNull()
...
+                }
+                finally
+                {
+                    Pipeline.LogMethodExit(nameof(QueueClient));
+                }
+            }
+        }
+        #endregion Exists
+
+        #region Delete If Exists
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Queues/src/QueueClient.cs]: NIT -Same thing I said for Create

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Queues/src/QueueClient.cs [hunk 3]
```diff
@@ -411,6 +417,352 @@ private void SetNameFieldsIfNull()
...
+            using (Pipeline.BeginLoggingScope(nameof(QueueClient)))
+            {
+                Pipeline.LogMethodEnter(
+                    nameof(QueueClient),
+                    message:
+                    $"{nameof(Uri)}: {Uri}");
+
+                try
+                {
+                    Response<QueueProperties> response = await QueueRestClient.Queue.GetPropertiesAsync(
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Queues/src/QueueClient.cs]: Nit - would it be easier to call `GetProperitesInternal` from here instead?

**Review by tg-msft — APPROVED** [review]
Looks good.

I wouldn't group parameters as part of this focused PR, but I agree that now might be a good time to start that work before you get any new long lived feature branches.  Maybe we could tie that to something like supporting x-ms-client-request-id per operation which we've been seeing requests about so customers get something for having to make the change?

File: sdk/storage/Azure.Storage.Queues/src/QueueClient.cs (hunk 1 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Queues/src/QueueClient.cs]: Fixed the spacing issue.

File: sdk/storage/Azure.Storage.Queues/src/QueueClient.cs (hunk 2 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Queues/src/QueueClient.cs]: Fixed the spacing issue.

File: sdk/storage/Azure.Storage.Queues/src/QueueClient.cs (hunk 3 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Queues/src/QueueClient.cs]: Yes, fixed.

**Pull request merged**